parsl/executors/high_throughput/executor.py

```diff
@@ -615,27 +615,27 @@ def scale_out(self, blocks=1):
             raise (ScalingFailed("No execution provider available"))
         r = []
         for i in range(blocks):
-            external_block_id = str(len(self.blocks))
+            block_id = str(len(self.blocks))
             try:
-                internal_block_id = self._launch_block(external_block_id)
-                self.blocks[external_block_id] = internal_block_id
-                self.block_mapping[internal_block_id] = external_block_id
-                r.append(external_block_id)
+                job_id = self._launch_block(block_id)
+                self.blocks[block_id] = job_id
+                self.block_mapping[job_id] = block_id
+                r.append(block_id)
             except Exception as ex:
-                self._fail_job_async(external_block_id,
-                                     "Failed to start block {}: {}".format(external_block_id, ex))
+                self._fail_job_async(block_id,
+                                     "Failed to start block {}: {}".format(block_id, ex))
         return r
 
-    def _launch_block(self, external_block_id: str) -> Any:
+    def _launch_block(self, block_id: str) -> Any:
```

Comment:
can the `Any` on this line turn into the correct type for a job id?

Reply:
I think the job id can be either `None` or `str`?

Reply:
hm I looked in provider_base.py and the ID coming back from a provider is defined to be `Any` there. So using `str` for a job ID anywhere is wrong.

Reply:
(or this PR could change the API for providers to insist that job IDs are str)